sdk/login_form/login_form_handler.js

```diff
@@ -1,37 +1,240 @@
-      .value.trim();
-
-    const password = document.getElementById('web-nexus-password-input')
-      .value.trim();
-
-    loginForm.parentElement.remove();
-
-    addUserOnline(username, password);
-  });
-}
-
-function createLoginForm(divId = undefined) {
-  fetch(loginFormUrl)
-    .then((result) => result.text())
-    .then((body) => {
-      if (divId) {
-        document.getElementById(divId).innerHTML = body;
-      } else {
-        const loginFormDiv = document.createElement('div');
-        loginFormDiv.innerHTML = body;
-        document.body.insertBefore(loginFormDiv, document.body.firstChild);
-      }
-
-      setUpLoginForm();
-    });
-}
-
-module.exports = { createLoginForm };
+const { addUserOnline } = require('../chat/chat_handlers');
+const bcrypt = require('bcryptjs');
+
+const loginFormUrl = 'http://localhost:8080/api/authentication/get-login-form';
+const registerFormUrl = 'http://localhost:8080/api/authentication/get-register-form';
+const backendLoginEndpoint = 'http://localhost:8080/api/authentication/login';
+const backendRegisterEndpoint = 'http://localhost:8080/api/authentication/register';
+
+const authEventHandler = {
+  onSuccessfulLoginHandler: defaultOnSuccessfulHandler,
+  onErrorHandler: () => {
+    alert('Wrong email or password!');
+  },
+  onServerErrorHandler: () => {
+    const url = 'http://localhost:63342/web-nexus/sdk/pages/internal-server-error.html';
+    window.location.replace(url);
+  }
+};
+
+const forms = {
+  loginForm: null,
+  registerForm: null
+};
+
+function initializeAuthenticationContainer(divId,
+  onSuccessfulCb = undefined,
+  onErrorCb = undefined,
+  onServerError = undefined)
+{
+  setUpLoginForm(divId);
+  setUpRegisterForm(divId);
+  setResponseEvents(onSuccessfulCb, onErrorCb, onServerError);
+}
+
+function setUpLoginForm(divId) {
+  fetch(loginFormUrl)
+    .then((result) => result.text())
+    .then((body) => {
+      if (divId) {
+        document.getElementById(divId).innerHTML = body;
+      } else {
+        const authContainerDiv = document.createElement('div');
+        authContainerDiv.innerHTML = body;
+        document.body.insertBefore(authContainerDiv, document.body.firstChild);
+      }
+
+      forms.loginForm = body;
+    })
+    .then(setUpLoginEvent)
+    .catch((error) => {
+      console.log(error);
+    });
+}
+
+function setUpLoginEvent() {
+  const loginForm = document.getElementById('web-nexus-auth-form');
+  loginForm.addEventListener('submit', function (event) {
+    event.preventDefault();
+    const email = document.getElementById('web-nexus-email-input')
+      .value.trim();
+    const password = document.getElementById('web-nexus-password-input')
+      .value.trim();
+    // const salt = bcrypt.genSaltSync(10);
+    // const hashedPassword = await bcrypt.hash(password, salt);
+
+    login(email, password)
+      .then((data) => {
+        authEventHandler.onSuccessfulLoginHandler(data);

```

Comment:
If the onSuccess Callback gets called here, wouldn't it be better to also call the onError Handlers here rather then in the login() function.

Reply:
Where exact did you mean?